File: code-optimization/readability/java/src/main/java/com/epam/aicode/java/DiscountAction.java
package com.epam.aicode.java;

public class DiscountAction {

    public Discount calculateDiscount(double price, String userType, boolean isHoliday) {
        double discountRate = 0;
        if (userType.equals("STUDENT")) {
            discountRate = price * 0.1;
        } else if (userType.equals("SENIOR")) {
            discountRate = price * 0.15;
        } else if (userType.equals("EMPLOYEE")) {
            discountRate = price * 0.2;
        }
        if (isHoliday) {
            discountRate += price * 0.05;
        }

        Discount discount = new Discount();
        discount.setRate(discountRate);
        discount.setUserType(userType);
        discount.setHoliday(isHoliday);

        return discount;
    }
}

File: code-optimization/readability/C#/VSCode/readability/DiscountAction.cs
public class DiscountAction
{
    public Discount CalculateDiscount(double price, string userType, bool isHoliday)
    {
        double discountRate = 0;
        if (userType.Equals("STUDENT"))
        {
            discountRate = price * 0.1;
        }
        else if (userType.Equals("SENIOR"))
        {
            discountRate = price * 0.15;
        }
        else if (userType.Equals("EMPLOYEE"))
        {
            discountRate = price * 0.2;
        }
        if (isHoliday)
        {
            discountRate += price * 0.05;
        }

        Discount discount = new Discount();
        discount.Rate = discountRate;
        discount.UserType = userType;
        discount.IsHoliday = isHoliday;

        return discount;
    }

}
